### PR TITLE
Add coverage support to spacejam

### PR DESCRIPTION
**Pull request by serut**
I worked a lot on spacejam to implement the coverage support using the lmieulet:meteor-coverage package.
The result is not perfect, but it's working at least !
Here is a repository that uses my own fork of spacejam (test, test --full-app, test-package using mocha and test-package using tinytest):
https://github.com/serut/meteor-coverage-app-exemple

Missing feature:
- merge the coverage between different type of tests.

Things missing to this pull request:
- Update the readme.md and the help.txt file

### CODE REVIEW

**Comment by serut** [comment]
Any feeling about this pull request ?

**Comment by minid33** [comment]
+1 :shipit:

File: lib/phantomjs-test-in-console-with-coverage.coffee [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+page.onCallback = (data) ->
+  ## Callback when sending and saving coverage
+  if data && data.err
+      console.log("coverage error: #{data.err}")
+      phantom.exit(7)
+  else
+     phantom.exit(0)
+
+
+checkingStatus = setInterval ->
```
Comment (minid33) [inline lib/phantomjs-test-in-console-with-coverage.coffee]: Quite a lot of cyclomatic complexity here.

**Comment by rbabayoff** [comment]
@serut, once we finish with windows support, we'll take a look at this.

**Comment by serut** [comment]
I've sometimes dependency injection issues when coverage is active that fails the test when phantomjs opens the page : 

>phantomjs: TypeError: undefined is not an object (evaluating 'Package['practicalmeteor:chai'].chai')
    http://localhost:4096/packages/practicalmeteor_loglevel.js?hash=96cd6887e3b954341d356b2fb1373d3149d292c4: 9

https://travis-ci.org/serut/meteor-coverage-app-exemple/builds/144737808#L4119

> phantomjs: TypeError: undefined is not an object (evaluating 'Package['practicalmeteor:loglevel'].loglevel')
    http://localhost:4096/packages/practicalmeteor_mocha.js?hash=0c7bae1a756c9691267ec935a8d6a4e081df9cf6: 9

https://travis-ci.org/serut/meteor-coverage-app-exemple/builds/144737808#L6036

Same issue on circle.ci : https://circleci.com/gh/serut/meteor-coverage-app-exemple/30

These errors happens only when the code is intrumented, so I will try to ignore all mocha stuff from instrumentation to see if these errors disappear...!

**Comment by serut** [comment]
This commit adds the ability to define what the test runner will do during its test.
Following possibilities are working, and you can mix them to do several task
`--coverage in_coverage` : import a `report.json` file which contains the coverage of the app, in order to merge it with the current coverage report
`--coverage out_coverage` : save the current coverage report in a `report.json` file
`--coverage out_lcovonly` : save the current coverage report in a `lcov.info` file. Such file can be sent to coveralls or codecov easily

So you can do:  
`--coverage 'in_coverage|out_lcovonly` : import a `report.json` file and save the current coverage report in a `lcov.info` file ;-) You will see the coverage of 2 types of test on the same report/HMI.

I propose to fork https://github.com/serut/meteor-coverage-app-exemple in the `/tests/apps` directory , execute all mocha test and check if the report.json and the lcov.info file exist. It would be a nice end-to-end test.

To conclude : 
- We can get app coverage from mocha tests
- No test yet
- This pull request refers an old branch
- Need readme and help.txt improvement

**Comment by vangorra** [comment]
I've been using the coverage fork for over a month now. It's working great. What must happen to get this merged?

**Comment by serut** [comment]
Thanks for testing @vangorra

- [ ] Integrate the PR *Windows suppport* #61 
      - [ ] Just need to fix tests 
- [ ] Remove this pull request because it does not point master
- [ ] Reports change from https://github.com/serut/spacejam/tree/windows-suppport-rc4 into https://github.com/serut/spacejam/tree/coverage-support

**Pull request closed without merging**